sdk/identity/azure-identity/azure/identity/_credentials/managed_identity.py

```diff
@@ -99,122 +83,3 @@ def get_token(self, *scopes, **kwargs):
-        config.headers_policy = HeadersPolicy(base_headers={"Metadata": "true"}, **kwargs)
-        config.logging_policy = NetworkTraceLoggingPolicy(**kwargs)
-        config.user_agent_policy = UserAgentPolicy(base_user_agent=USER_AGENT, **kwargs)
-
-        return config
-
-    # given RetryPolicy's implementation, these settings most closely match the documented guidance for IMDS
-    # https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#retry-guidance
-    _retry_settings = {
-        "retry_total": 5,
-        "retry_status": 5,
-        "retry_backoff_factor": 4,
-        "retry_backoff_max": 60,
-        "retry_on_status_codes": [404, 429] + list(range(500, 600)),
-    }
-
-
-class ImdsCredential(_ManagedIdentityBase, GetTokenMixin):
-    """Authenticates with a managed identity via the IMDS endpoint.
-
-    :keyword str client_id: ID of a user-assigned identity. Leave unspecified to use a system-assigned identity.
-    """
-
-    def __init__(self, **kwargs):
-        # type: (**Any) -> None
-        super(ImdsCredential, self).__init__(endpoint=Endpoints.IMDS, client_cls=AuthnClient, **kwargs)
-        self._endpoint_available = None  # type: Optional[bool]
-
-    def _acquire_token_silently(self, *scopes):
-        # type: (*str) -> Optional[AccessToken]
-        return self._client.get_cached_token(scopes)
-
-    def _request_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
-        # type: (*str, **Any) -> AccessToken
-        if self._endpoint_available is None:
-            # Lacking another way to determine whether the IMDS endpoint is listening,
-            # we send a request it would immediately reject (missing a required header),
-            # setting a short timeout.
-            try:
-                self._client.request_token(scopes, method="GET", connection_timeout=0.3, retry_total=0)
-                self._endpoint_available = True
-            except HttpResponseError:
-                # received a response, choked on it
-                self._endpoint_available = True
-            except Exception:  # pylint:disable=broad-except
-                # if anything else was raised, assume the endpoint is unavailable
-                self._endpoint_available = False
-                _LOGGER.info("No response from the IMDS endpoint.")
-
-        if not self._endpoint_available:
-            message = "ManagedIdentityCredential authentication unavailable, no managed identity endpoint found."
-            raise CredentialUnavailableError(message=message)
-
-        if len(scopes) != 1:
-            raise ValueError("This credential requires exactly one scope per token request.")
-
-        resource = scopes[0]
-        if resource.endswith("/.default"):
-            resource = resource[: -len("/.default")]
-        params = dict({"api-version": "2018-02-01", "resource": resource}, **self._identity_config)
```

Comment:
Is this scopes restriction still being applied after the refactor?

Reply:
Yes, `ManagedIdentityClient.request_token` calls [`_scopes_to_resource`](https://github.com/Azure/azure-sdk-for-python/blob/355c45cc5b4f3861694eca6c0c7f662ad5cc5350/sdk/identity/azure-identity/azure/identity/_internal/__init__.py#L55), which is a copy of this.